### PR TITLE
Feature/c8 policy changes prototype

### PR DESCRIPTION
**Pull request by JamesCuttsHmcts**
## JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/PRL-6480
https://tools.hmcts.net/jira/browse/PRL-6427
https://tools.hmcts.net/jira/browse/PRL-6481
https://tools.hmcts.net/jira/browse/PRL-6500
https://tools.hmcts.net/jira/browse/PRL-6508
https://tools.hmcts.net/jira/browse/PRL-6503
https://tools.hmcts.net/jira/browse/PRL-6504
https://tools.hmcts.net/jira/browse/PRL-6502

### Change description ###

Adds C8 refuge screens for applicant and other people in the C100 application journey and for the CA/DA confirm your contact details journey.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/steps/common/refuge/c8-already-uploaded/content.ts [hunk 1]
```diff
@@ -0,0 +1,120 @@
+import { YesOrNo } from '../../../../app/case/definition';
```
Comment (vivek-sekhar) [inline src/main/steps/common/refuge/c8-already-uploaded/content.ts:1]: change folder name to refuge-document-already-uploaded

File: src/main/steps/common/refuge/staying-in-refuge/content.ts [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    },
+  },
+};
+
+const cy: typeof en = {
+  title: 'Staying in a refuge (welsh)',
+  refuge:
+    'A refuge is a secure place for people and their children to stay when they are escaping domestic abuse. It provides a space to feel safe and supported. (welsh)',
+  citizensAdvice:
+    'Find out more about refuges at <a href="https://www.citizensadvice.org.uk/" class="govuk-link" target="_blank" aria-label="This link will open in a new tab for Citizen\'s Advice">Citizen\'s Advice (opens in a new tab).</a> (welsh)',
```
Comment (vivek-sekhar) [inline src/main/steps/common/refuge/staying-in-refuge/content.ts:36]: make sure to add welsh content for aria-label

File: src/main/steps/common/refuge/staying-in-refuge/postController.ts [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    const { onlyContinue, saveAndComeLater, ...formFields } = req.body;
+    const userCase = req.session.userCase;
+    const id = req.params.id;
+    const { citizenUserLivingInRefuge } = formFields;
+
+    if (!_.isEmpty(citizenUserLivingInRefuge)) {
+      if (req?.originalUrl?.startsWith(C100_URL)) {
+        const c100Person = getPeople(userCase).find(person => person.id === id)!;
+        if (c100Person.partyType === PartyType.APPLICANT) {
```
Comment (vivek-sekhar) [inline src/main/steps/common/refuge/staying-in-refuge/postController.ts:33]: refactor if and else

File: src/main/steps/common/refuge/staying-in-refuge/routeGuard.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+export const routeGuard = {
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
+  get: async (req: AppRequest, res: Response, next: NextFunction) => {
+    const userCase = req.session.userCase;
+    const id = req.params.id;
+
+    if (req?.originalUrl?.startsWith(C100_URL)) {
+      const c100Person = getPeople(userCase).find(person => person.id === id)!;
+      if (c100Person.partyType === PartyType.APPLICANT) {
```
Comment (vivek-sekhar) [inline src/main/steps/common/refuge/staying-in-refuge/routeGuard.ts:17]: refactor if and else

File: src/main/steps/common/refuge/upload-refuge-doc/postController.ts [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+        files: [files?.['c8RefugeDocument']],
+      });
+
+      if (response.status !== 'Success') {
+        req.session.errors = handleError(req.session.errors, 'uploadError', 'c8RefugeDocument', true);
+        return;
+      }
+
+      if (C100rebuildJourney) {
+        if (c100Person.partyType === PartyType.APPLICANT) {
```
Comment (vivek-sekhar) [inline src/main/steps/common/refuge/upload-refuge-doc/postController.ts:67]: refactor if and else

File: src/main/steps/common/refuge/utils.ts [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        req.session.userCase.oprs_otherPersons
+      ) as C100RebuildPartyDetails[];
+    }
+  }
+};
+
+export const getC8DocumentForC100 = (id: string, caseData: CaseWithId, person: People): Document => {
+  let c8Document;
+
+  if (person.partyType === PartyType.APPLICANT) {
```
Comment (vivek-sekhar) [inline src/main/steps/common/refuge/utils.ts:84]: refactor if and else

File: src/main/steps/common/refuge/utils.ts [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+      req.session.save(() => {
+        res.redirect(redirectUrl);
+      });
+    }
+  }
+};
+
+export const deleteC100RefugeDoc = (req: AppRequest, caseData: CaseWithId, id: string): void => {
+  const c100Person = getPeople(caseData).find(person => person.id === id)!;
+  if (c100Person.partyType === PartyType.APPLICANT) {
```
Comment (vivek-sekhar) [inline src/main/steps/common/refuge/utils.ts:60]: refactor if and else

File: src/main/app/case/case.ts [hunk 8]
```diff
@@ -248,6 +248,7 @@ export const formFieldsToCaseMapping: Partial<Record<keyof Case, keyof CaseData>
   finalServedApplicationDetailsList: 'finalServedApplicationDetailsList',
   newChildDetails: 'newChildDetails',
   citizenNotifications: 'citizenNotifications',
+  citizenUserLivingInRefuge: 'citizenUserLivingInRefuge',
```
[the remark this answers is not in the document]
Comment (hcarslaw) [reply src/main/app/case/case.ts]: removed

File: src/main/app/case/definition.ts [hunk 9]
```diff
@@ -856,6 +860,7 @@ export interface CaseData {
   citizenUserManualAddressCounty?: string;
   c100Applicants?: C100Applicant;
   citizenUserManualAddressPostcode?: string;
+  citizenUserLivingInRefuge?: string;
```
[the remark this answers is not in the document]
Comment (hcarslaw) [reply src/main/app/case/definition.ts]: removed

File: src/main/app/case/definition.ts [hunk 10]
```diff
@@ -895,6 +900,7 @@ export interface CaseData {
   citizenApplicationPacks?: CitizenApplicationPacks[];
   finalServedApplicationDetailsList?: ServedApplicationDetails[];
   citizenNotifications?: CitizenNotification[];
+  livingInRefuge?: string;
```
[the remark this answers is not in the document]
Comment (hcarslaw) [reply src/main/app/case/definition.ts]: removed

File: src/main/modules/reasonable-adjustments/navigationController.ts [hunk 11]
```diff
@@ -125,7 +125,9 @@ export class ReasonableAdjustementsNavigationController {
 
     const partyType = getCasePartyType(caseData, req.session.user.id);
     const isC100Journey = currentPageUrl.startsWith(C100_URL);
-    const currentPageUrlParts = currentPageUrl.split('/');
+    const currentPageUrlPartsWithoutToken = currentPageUrl.split('?');
```
[the remark this answers is not in the document]
Comment (hcarslaw) [reply src/main/modules/reasonable-adjustments/navigationController.ts]: removed

File: src/main/steps/common/refuge/keeping-details-safe/content.ts [hunk 12]
```diff
@@ -81,22 +81,22 @@ export const generateContent = (content: CommonContent): PageContent => {
     ...translations,
     title: C100rebuildJourney
       ? interpolate(translations.c100.title, {
-          name: `${c100Person.firstName} ${c100Person.lastName}`,
+          name: `${c100Person?.firstName} ${c100Person?.lastName}`,
```
Comment (vivek-sekhar) [inline src/main/steps/common/refuge/keeping-details-safe/content.ts:84]: if there is no other person does it display undefined or blank for names?

**Review by vivek-sekhar — APPROVED** [review]

File: src/main/steps/common/refuge/keeping-details-safe/content.ts (hunk 12 above)
Comment (hcarslaw) [reply src/main/steps/common/refuge/keeping-details-safe/content.ts:84]: It displays undefined

**Pull request merged**